### DISCUSSION TITLE
fmax

### DISCUSSION THREAD

**Discussion by Navids94** [geodynamics/sw4#200] [Q&A]
Hi @andersp 
It is unclear to me which fmax should be considered for simulations up to 5Hz (fmax in source command(fmax=2.5f0), fmax=2.5*fc2 in prefilter, fmax=Vsmin/h/ppw or maxfreq in attenuation command)
I used following input for simulation 5Hz:

grid x=47878 y=21708 z=28000 h=40
time t=50
supergrid gp=30
pfile filename=pfile_Kinburn.ppmod style=cartesian
attenuation nmech=3 phasefreq=1 maxfreq=5.0
refinement zmax=175
refinement zmax=325
refinement zmax=625
source x=45878 y=2000 z=14500 strike=306 rake=94 dip=41 m0=1.32e+16 type=Liu freq=12.8 t0=5
prefilter fc2=5.0 type=lowpass passes=2 order=2

Based on the manual, fmax=Vsmin/h/ppw.  Vsmin at soil is 198m/s and at rock is 3470m/s , so:
Soil:  fmax=198/5m/8ppw=4.95Hz
Rock: fmax=3470/10m/8=43.37Hz

According to examples (Alumrock.in), fmax is equal to corner frequency (fc2 prefilter command) x 2.5, so based on this, I would have:
fmax=2.5*5=12.5Hz

Also, freq=12.8 so we have:
f0=freq/2pi=12.8/2pi=2.03Hz and fmax~2.5f0=5Hz

I would appreciate it if you could tell me for simulation up to 5Hz, I should use fc2=5.0 and maxfreq=5.0 or not.

**Comment by andersp** [comment]
Hello SW4 user,
The fmax in the attenuation command determines the coefficients in the attenuation model such that Qp and Qs are approximately constant functions of frequency, for frequencies between 0.01*fmax and fmax. If you resolve frequencies to 5 Hz, I would set fmax=10 Hz.
The fc2 variable in the prefilter command specifies the corner frequency in the Butterworth low-pass filter. In the 5 Hz case, it is appropriate to set it to fc2=5.0 Hz.
To trigger motions to 5 Hz, you need to set the time function in the source command appropriately. You are using a Liu function with angular corner frequency of about 4*pi rad/s, corresponding to 2 Hz. I would reduce the rise time in the Liu function by increasing the angular frequency to freq = 10*pi rad/s.
Good luck!

**Reply by Navids94** [reply]
Thank you very much for your reply. In order to resolve 5Hz, I used the following input based on your suggestion:
grid x=47878 y=21708 z=28000 h=40
attenuation nmech=3 phasefreq=1 maxfreq=10
refinement zmax=175
refinement zmax=325
refinement zmax=625
source x=45878 y=2000 z=14500 strike=306 rake=94 dip=41 m0=1.32e+16 type=Liu freq=31.41 t0=2
prefilter fc2=5.0 type=lowpass passes=2 order=2

According to the manual max frequency which we can simulate is:
fmax=Vsmin/PPW/h
In the soil Vsmin=198m/s and h=5m so:
fmax=198/8/5~5Hz
There is just one question I would like to ask. Does the sw4 automatically define ppw=8? If sw4 considers more than ppw=8, I can't get a resolution up to 5Hz.
Thank you so much for taking the time! I really appreciate it and I'm grateful for your help.

**Reply by andersp** [reply]
SW4 will calculate on the grid you give, no matter what value Vsmin/h/fmax has. At startup, SW4 reports Vsmin/h in each grid block. By dividing these numbers by fmax (5 Hz in your case), you need to check that PPW = Vsmin/h/fmax is at least near 8 in all grids. If PPW is smaller than 8 in any grid, you need to adjust the refinement levels to better optimize the grid for your material model. Note that this check can be done without any time stepping. You can also get a rough estimate of where the refinement levels need to be on a courser grid, e.g. h=80 should give adequate resolution for a 2.5 Hz simulation.

**Reply by Navids94** [reply]
Thank you so much for your reply. In each grid block, I received the following Vsmin/h. By dividing 5Hz, I obtain PPW ~ 8 for soil and 57 for rock. As Vsmin rock is around 3470m/s below 150m depth, I cannot get ppw=8 for rock and sw4 gives me a larger ppw.
g=0, h=4.000000e+01, minVs/h=86.25 (Cartesian)
g=1, h=2.000000e+01, minVs/h=173.5 (Cartesian)
g=2, h=1.000000e+01, minVs/h=285.171 (Cartesian)
g=3, h=5.000000e+00, minVs/h=39.6 (Cartesian)
Thank you so much for taking the time! 
Best regards,
